### ISSUE TITLE
Product Engineering Review: Energy Consumption Forecasting

### ISSUE THREAD

**Issue by ForbiddenDevil** [Teradata/jupyter-demos#528]

**Comment by ForbiddenDevil** [comment]
Changes incorporated. PR: https://github.com/Teradata/jupyter-demos/pull/574


> **Reviewer 1 comments:**
> 
> * lineplot using seaborn **_--- Fixed in pass6_**
> * sklearn minmaxscaler used and transform used scaler have used **_--- Fixed in pass6_**
> * subplot used using seaborn **_--- Fixed in pass6_**
> * lot of places to_pandas used **_--- Fixed in pass6_**
> * we have imported Linearregression, MinMaxscaler which gives customers wrong info that teradataml doesn't support this function. **_--- Fixed in pass6_**
> 
> **Reviewer 1 suggestions:**
> 
> * teradataml plot function supports both lineplot and subplots  **_--- Done in pass6_**
> * we can use minmaxscaler and transform of val functions **_--- Done in pass6_**
> * we can avoid using to_pandas and used teradataml functions example we have used pandas here
>   `normalize_value = int(df.tail(24).mean()['consumption']) normalize_value`
>   we can use teradataml functionality here
>   `normalize_value=int(df.tail(24).mean().select('consumption').squeeze())` **_--- Thank you for the suggestion. I'll include this_**
> 
> **Reviewer 2 comments:**
> 
> * user suppressed filter warnings _**--- This is done to keep the notebook clean. This is the standard followed for all notebooks**_
> * sklearn Pipeline imported but not used I think _**--- Removed the import**_
> * Used sklearn's LinearRegression, RandomForestRegressor and MinMaxScaler _**--- This is intended to simulate model being trained outside Vantage**_
> * Section 2 talks about "Only the minimal amount of data required is copied to the client" but used to_pandas() on all rows for plotting. **_--- Fixed in pass6_**
> * Used MinMaxScaler and its fit_transform() to scale the data and then plotting is done (esp. lineplot) **_--- Fixed in pass6_**
> * Used boxplot again on the data pulled to client. **_--- Fixed in pass6_**
> * Made good use of VAL's MinMaxScalar, OneHotEncoder, Retain and Transform functions. _**--- Thanks**_
> * Used copy_to_sql to split train and test data by 168 rows _**--- There is a specific requirement to split data in a sorted manner. This is not supported by sampling methods in teradataml**_
> * Visualized results of consumption vs prediction using subplots of LinearRegression, RandomForestRegressor models. **_--- Fixed in pass6_**
> * If the user intention is to demo use of BYOM and PMMLPredict, this should be fine. Otherwise, user can use upcoming OpensourceML for training sklearn models without pulling data from Vantage. _**--- The intention is to show BYOM**_
> 
> **Reviewer 2 suggestions:**
> 
> * We don’t suggest to skip warnings unless it is necessary. _**--- This is done to keep the notebook clean. This is the standard followed for all notebooks**_
> * can remove imported sklearn Pipeline if not used. _**--- Removed the unused import**_
> * Can use OpensourceML’s sklearn's LinearRegression, RandomForestRegressor and MinMaxScaler _**--- The intention is to show BYOM**_
> * Avoid pulling data to to_pandas(). Use it only when necessary. **_--- Fixed in pass6_**
> * Both can be accomplished by MinMaxScalar of VAL (or upcoming OpensourceML MinMaxScaler) and teradataml Plot. **_--- Fixed in pass6_**
> * See if teradataml Plot supports this. **_--- Removed box plot entirely_**
> * Made good use of VAL's MinMaxScalar, OneHotEncoder, Retain and Transform functions. Kudos here _**--- Thanks**_
> * Instead user can use sample for splitting the data and use those dataframes. **--- There is a specific requirement to split data in a sorted manner. This is not supported by sampling methods in teradataml**_
> * teradataml Plot can be used without pulling data. **_--- Fixed in pass6_**

**Comment by shilpa-nalkande** [comment]
Review completed and executed successfully

**Comment by ForbiddenDevil** [comment]
Merged. Can be moved to "Done" phase.
